src/test/kotlin/com/nextstep/racingcar/domain/CarTest.kt

```diff
@@ -1,34 +1,66 @@
 package com.nextstep.racingcar.domain
 
-import com.nextstep.racingcar.domain.rules.NumberGenerator
-import com.nextstep.racingcar.domain.rules.NumberMoveRule
+import com.nextstep.racingcar.domain.Movement.MOVE
+import com.nextstep.racingcar.domain.Movement.NONE
+import com.nextstep.racingcar.domain.rules.MoveRule
+import io.kotest.assertions.throwables.shouldThrow
 import io.kotest.core.spec.style.StringSpec
 import io.kotest.data.forAll
 import io.kotest.data.row
 import io.kotest.matchers.collections.shouldContainExactly
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.throwable.shouldHaveMessage
 import io.mockk.every
 import io.mockk.mockk
 import io.mockk.verify
 
 class CarTest : StringSpec({
-    val numberGenerator = mockk<NumberGenerator>()
-    val moveRule = NumberMoveRule(numberGenerator)
 
-    "Car#move returns Movement Enum meaning thee car moved or not" {
+    "Car#constructor exception test: car name should be shorter than 6 letters and not empty" {
         forAll(
-            row(0, Movement.NONE),
-            row(3, Movement.NONE),
-            row(4, Movement.MOVE),
-            row(9, Movement.MOVE)
-        ) { number, movement ->
-            val car = Car()
-            every { numberGenerator.generate() } returns number
+            row("long name"),
+            row(""),
+            row("    ")
+        ) { name ->
+            shouldThrow<IllegalArgumentException> {
+                Car(name)
+            } shouldHaveMessage "car name should be shorter than 6 letters and not empty"
+        }
+    }
+
+    "If car is called, car delegate moveRule to move" {
+        val car = Car("sujin")
+        val moveRule = mockk<MoveRule>()
+        every { moveRule.move() } returns MOVE
+
+        car.move(moveRule)
+
+        verify(exactly = 1) { moveRule.move() }
```

Comment:
강의에서 배운 것 처럼, 아래 스니펫과 같이 모킹 없이 테스트가 가능하겠네요 :)
```suggestion
        // given
        val car = Car("sujin")

        // when
        car.move { MOVE }
        
        // then
        car.getLocation() shouldBe 1
```

Reply:
오???? 그러네요! 감사합니다 :)